streams/src/main/java/org/apache/kafka/streams/processor/internals/AssignedTasks.java

```diff
@@ -331,26 +331,16 @@ void closeNonAssignedSuspendedTasks(final Map<TaskId, Set<TopicPartition>> newAs
 
     void close(final boolean clean) {
         final AtomicReference<RuntimeException> firstException = new AtomicReference<>(null);
-        for (final T task : allTasks()) {
-            try {
-                task.close(clean, false);
```

Comment:
Why not leaf the code "as-is" and change this line to:
```
if (suspended.values().contains(task)) {
    task.closeSuspended(clean, false, firstException);
} else {
    task.close(clean, false);
}
```

Reply:
Absolutely! Will do.